File: MergeSort/MergeSort.java
public class MergeSort {
	
	private static void merge(Comparable[] a) {
		//same as before??
	}

	private static void sort(Comparable[] a, Comparable[] aux, int lo, int hi) {
		if (hi <= lo) return;
		int mid = lo + (hi - lo) / 2;
		sort(a, aux, lo, mid);
		sort(a, aux, mid+1, hi);
		merge(a, aux, lo, mid, hi);
	}

	public static void sort(Comparable[] a) {
		aux = new Comparable[a.length];
		sort(a, aux, 0, a.length - 1);
	}
}

File: Javascript/MergeSort.js
var mergeA = [5, 10, 3, 6, 1, 0, 56, 32, 90, 34];
count = 0;

var MergeSort = function(array) {
	console.log(count++);

	if (array.length < 2) {
		return array;
	}

	var mid = Math.floor(array.length/2);

	var left=array.slice(0,mid);
	var right=array.slice(mid);
	console.log(left);
	console.log(right);

	return merge(MergeSort(left), MergeSort(right));
}

var merge = function(left,right) {

	var aux = [],
	leftpart = 0,
	rightpart = 0;

	while (leftpart < left.length && rightpart < right.length) {
		if (left[leftpart] < right[rightpart]) {
			aux.push(left[leftpart++]);
		}
		else {
			aux.push(right[rightpart++]);
		}
	}
	return aux.concat(left.slice(leftpart)).concat(right.slice(rightpart));
}

var Msort = function(array) {

	var count = 0;

	MergeSort(array, aux, 0, array.length - 1)
}

var less = function(v1, v2) {
	if (v1 < v2) {
		return true;
	}
	else {
		return false;
	}
}

var exch = function(array,v1,v2) {
	var swap = array[v1];
	array[v1] = array[v2];
	array[v2] = swap;
}

console.log(MergeSort(mergeA));